src/librustdoc/html/static/main.js

```diff
@@ -1369,13 +1369,17 @@
 
         function search(e) {
             var params = getQueryStringParams();
-            var query = getQuery(document.getElementsByClassName('search-input')[0].value);
+            var search_input = document.getElementsByClassName('search-input')[0];
+            var query = getQuery(search_input.value);
 
             if (e) {
                 e.preventDefault();
             }
 
             if (!query.query || query.id === currentResults) {
+                if (query.query.length > 0) {
```

Comment:
Will this bite us if this block is entered from `!query.query`? Can `query.query` be undefined here?

Reply:
Based on [this code](https://github.com/rust-lang/rust/blob/master/src/librustdoc/html/static/main.js#L1056), I'd say no. This is why I'm complaining about such conditions: it's difficult to understand what we're checking. I think the "real" condition here is `if (query.query.length === 0 || query.id === currentResults) {`.